addons/l10n_ch/models/res_bank.py

```diff
@@ -133,12 +133,12 @@ def _l10n_ch_get_qr_vals(self, amount, currency, debtor_partner, free_communicat
             '',                                                   # Ultimate Creditor Country
             '{:.2f}'.format(amount),                              # Amount
             currency.name,                                        # Currency
-            'K',                                                  # Ultimate Debtor Address Type
+            'S',                                                  # Ultimate Debtor Address Type
             debtor_partner.commercial_partner_id.name[:70],       # Ultimate Debtor Name
-            debtor_addr_1,                                        # Ultimate Debtor Address Line 1
-            debtor_addr_2,                                        # Ultimate Debtor Address Line 2
-            '',                                                   # Ultimate Debtor Postal Code (not to be provided for address type K)
-            '',                                                   # Ultimate Debtor Postal City (not to be provided for address type K)
+            debtor_addr['street_name'][:70],                      # Ultimate Debtor Street Name
+            debtor_addr['street_number'][:16],                    # Ultimate Debtor Building Number
+            debtor_partner.zip,                                   # Ultimate Debtor Postal Code
+            debtor_partner.city[:35],                             # Ultimate Debtor Town
             debtor_partner.country_id.code,                       # Ultimate Debtor Postal Country
```

Comment:
Typo in comment part.
```suggestion
            debtor_partner.country_id.code,                       # Ultimate Debtor Country
```